crates/autopilot/src/run_loop.rs

```diff
@@ -429,6 +452,48 @@ impl RunLoop {
         Ok(())
     }
 
+    /// Tries to find a `settle` contract call with calldata ending in `tag`.
+    ///
+    /// Returns None if no transaction was found within the deadline or the task
+    /// is cancelled.
+    async fn wait_for_settlement_transaction(
+        &self,
+        tag: &[u8],
+        max_blocks_wait: u64,
+    ) -> Result<Option<Transaction>> {
+        let start_offset = MAX_REORG_BLOCK_COUNT;
+        let current = self.eth.current_block().borrow().number;
+        let start = current.saturating_sub(start_offset);
+        let deadline = current.saturating_add(max_blocks_wait);
+        tracing::debug!(%current, %start, %deadline, ?tag, "waiting for tag");
+        let mut seen_transactions: HashSet<H256> = Default::default();
+        loop {
+            if self.eth.current_block().borrow().number > deadline {
+                break;
+            }
+            let mut hashes = self
+                .persistence
+                .recent_settlement_tx_hashes(start..deadline + 1)
+                .await?;
```

Comment:
A bit inconsistent that a DB error aborts the whole thing while a node error will just lead to a log.
I think logging the DB error (maybe with `.inspect_err()` would be fine).